includes/class-fibpg-api-payment.php

```diff
@@ -57,8 +56,24 @@ public static function create_qr_code($order, $access_token)
             wc_add_notice(esc_html__('Something went wrong!', 'fib-payments-gateway'), 'error');
             return;
         }
-        $_SESSION['payment_id'] = sanitize_text_field($response_data['paymentId']);
-        return filter_var($response_data['qrCode'], FILTER_SANITIZE_URL);
+        $qr_code = $response_data['qrCode'];
+        $business_app_link = $response_data['businessAppLink'];
+        $corporate_app_link = $response_data['corporateAppLink'];
+        $personal_app_link = $response_data['personalAppLink'];
+
+        $readable_code = $response_data['readableCode'];
+
+        update_post_meta($order->get_id(), '_fib_payment_id', $response_data['paymentId']);
```

Comment:
Change it to be the same as other response data assign it to a variable or we need only `$qr_code` to be variable because of function return but other variables are used one time inside the function so can directly `update_post_meta` using `$response_data`

Reply:
Thanks for the comment, both ways work but i have used the variables for better readability of the code. I will make sure put the paymentId into a variable as well.